packages/language-server/src/plugins/PluginHost.ts

```diff
@@ -334,6 +335,32 @@ export class PluginHost implements LSProvider, OnWatchFileChanges {
         );
     }
 
+    async getSelectionRanges(
```

Comment:
Could you add a small comment above the function briefly explaining why this is a special case and the ranges have to be accumulated like this?